File: 35. Search Insert Position Easy/solution.py
def search_insert(nums: list[int], target: int) -> int:
    left, right = 0, len(nums) - 1
    while left <= right:
        mid = (left + right) // 2
        if nums[mid] == target:
            return mid
        if nums[left] < target:
            left += 1
        else:
            right -= 1
    return left

print(search_insert([1,3,5,6], 5))
print(search_insert([1,3,5,6], 2))
print(search_insert([1,3,5,6], 7))

File: 35. Search Insert Position Easy/solution.ts
function searchInsert(nums: number[], target: number): number {
    let [left, right] = [0, nums.length];
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        if (nums[mid] === target) return mid;

        if (nums[left] < target) {
            left++;
        } else {
            right--;
        }
    }

    return left;
}

console.log(searchInsert([1,3,5,6], 5))
console.log(searchInsert([1,3,5,6], 2))
console.log(searchInsert([1,3,5,6], 7))